src/wincurse.cpp

```diff
@@ -627,9 +625,40 @@ int curses_getch(WINDOW* win)
         ShowCursor(false);
     }
 
-    return lastchar;
+    previously_pressed_key = 0;
+    input_event rval;
+    if( lastchar == ERR ) {
+        if( input_timeout > 0 ) {
+            rval.type = CATA_INPUT_TIMEOUT;
+        } else {
+            rval.type = CATA_INPUT_ERROR;
+        }
+    } else {
+        if( lastchar == 127 ) { // == Unicode DELETE
+            previously_pressed_key = KEY_BACKSPACE;
+            return input_event( KEY_BACKSPACE, CATA_INPUT_KEYBOARD );
+        }
+        rval.type = CATA_INPUT_KEYBOARD;
+        rval.text = UTF32_to_UTF8( lastchar );
```

Comment:
fixing in merge, utf32_to_utf8(), probably left caps lock on after typing previous line or something.  Also #include "catacharset.h".